Kernel/Arch/x86_64/ISABus/I8042Controller.cpp

```diff
@@ -6,17 +6,37 @@
 
 #include <Kernel/Arch/Delay.h>
 #include <Kernel/Arch/x86_64/IO.h>
-#include <Kernel/Arch/x86_64/ISABus/HID/PS2KeyboardDevice.h>
-#include <Kernel/Arch/x86_64/ISABus/HID/PS2MouseDevice.h>
 #include <Kernel/Arch/x86_64/ISABus/HID/VMWareMouseDevice.h>
 #include <Kernel/Arch/x86_64/ISABus/I8042Controller.h>
+#include <Kernel/Bus/SerialIO/Device.h>
 #include <Kernel/Devices/HID/KeyboardDevice.h>
 #include <Kernel/Devices/HID/MouseDevice.h>
+#include <Kernel/Devices/HID/PS2/KeyboardDevice.h>
+#include <Kernel/Devices/HID/PS2/MouseDevice.h>
 #include <Kernel/Devices/HID/ScanCodeEvent.h>
 #include <Kernel/Sections.h>
 
 namespace Kernel {
 
+#define IRQ_FIRST_PORT 1
+#define IRQ_SECOND_PORT 12
+
+UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<I8042ControllerIRQHandler>> I8042ControllerIRQHandler::try_create(I8042Controller const& controller, u8 irq_number)
+{
+    return adopt_nonnull_own_or_enomem(new I8042ControllerIRQHandler(controller, irq_number));
+}
+
+bool I8042ControllerIRQHandler::handle_irq(RegisterState const&)
+{
+    return m_controller->handle_irq({}, interrupt_number());
+}
+
+UNMAP_AFTER_INIT I8042ControllerIRQHandler::I8042ControllerIRQHandler(I8042Controller const& controller, u8 irq_number)
+    : IRQHandler(irq_number)
+    , m_controller(controller)
+{
+}
+
```

Comment:
This change looks good to me, I like the simple IRQ handler logic compared to the previous `KeyboardDevice/MouseDevice` receives interrupt -> calls into `I8042Controller` -> calls back into `KeyboardDevice/MouseDevice`.

Reply:
Yeah, it looks nice now :)